### ISSUE TITLE
Remove async_trait macro in favor of AFIT & RPITIT

### ISSUE THREAD

**Issue by theangryangel** [theangryangel/insim.rs#123]
Requires minimum rust version 1.75.0

**Comment by theangryangel** [comment]
This is now unlocked after #166

**Comment by theangryangel** [comment]
https://blog.rust-lang.org/2024/09/05/impl-trait-capture-rules.html

Worth potentially holding until 2024 edition

**Comment by theangryangel** [comment]
Unnecessary as a result of #195, which I now expect to continue with and merge into main.
